pkg/urbit/jets/e/de_json.c

```diff
@@ -0,0 +1,959 @@
+/* j/5/de_json.c
+**
+*/
+#include "all.h"
+
+/* most of the code below is copied from the Public Domain JSON Parser for C
+*  pdjson: https://github.com/skeeto/pdjson
+*/
```

Comment:
Yeah, I was hoping for something like this rather than incorporating and "urbitizing" the source code. List of all changes below, non-cosmetic changes bolded:
1. changed types to match u3
2. renamed variables to match u3 style
3. renamed helper functions
4. **removed all input options other than byte buffer**
5. **removed errmsg buffer which tracked the exact cause of parsing failure**
6. **added a `u3_noun` to `json_stack` to store intermediate collections (list/map) while still parsing input**
7. **merged `json_stream` and `json_source` initialization functions**
8. removed many functions that are useful for a generic JSON parser, but not useful for our usage
Only change 6 would require a non-trivial amount of work to redo if we treat pdjson as an external library, and it would be quite simple.

Reply:
While changing lists to use `u3i_defcons`, I replaced the stack. So, now the jet can easily be modified to only make calls to the unmodified pdjson code.